core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/Block.scala

```diff
@@ -38,16 +41,20 @@ class Block(val name: Name,
             val rules: NonEmptyList[Rule])
   extends Logging {
 
-  def execute(requestContext: RequestContext): BlockResultWithHistory = {
+  def execute(requestContext: RequestContext)
+             (implicit loggingContext: LoggingContext): BlockResultWithHistory = {
     processRules(rules.toList, requestContext)
   }
 
-  def executeUserMetadataRuleOnly(requestContext: RequestContext): BlockResultWithHistory = {
+  def executeUserMetadataRuleOnly(requestContext: RequestContext)
+                                 (implicit loggingContext: LoggingContext): BlockResultWithHistory = {
     processRules(rules.collect { case r: UserMetadataRelatedRule => r }, requestContext)
   }
 
   private def processRules(selectedRules: List[Rule],
-                           requestContext: RequestContext): BlockResultWithHistory = {
+                           requestContext: RequestContext)
+                          (implicit loggingContext: LoggingContext): BlockResultWithHistory = {
+    implicit val showHeader: Show[Header] = ObfuscatedHeaderShowFactory.create(loggingContext.obfuscatedHeaders)
```

Comment:
please move creation of show instance close to Show[RequestContext] instance - seems that this code is only interested in using the showHeader